openapi.yaml

```diff
@@ -373,62 +465,122 @@ components:
+        feedbackText:
+          type: string
+          description: Text content of the feedback
+          nullable: true
+        externalThreadId:
           type: string
-          description: ID of the specific message being rated
-        feedback:
-          $ref: '#/components/schemas/Feedback'
-        suggestedChanges:
+          description: External ID of the thread containing the target message
+        externalMessageId:
           type: string
-          description: Optional proposed changes to the message content
+          description: ID of the specific message for feedback. If not provided, feedback will be attached to the thread's last message
           nullable: true
-        user:
-          $ref: '#/components/schemas/UserInfo'
+        projectId:
+          type: integer
+          description: Recommended - ID of the project this thread belongs to. Helps ensure correct thread identification
+          nullable: true
+        externalUser:
+          $ref: '#/components/schemas/CreateExternalUserRequest'
+          nullable: true
+          description: External user information for associating feedback with a user
+        attributes:
+          type: object
+          description: |
+            Key-value pairs where:
+            - key: attribute name (must match an existing attribute in the project)
+            - value: option name (must match an existing option for that attribute)
+          additionalProperties:
+            type: string
+          example:
+            Quality: "Wonderful"
+            Relevance: "High"
+      required:
+        - feedbackType
+        - externalThreadId
+
+    ExternalFeedbackMessageAssociation:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int32
+          description: Unique identifier for the message association
+        messageId:
+          type: integer
+          format: int32
+          description: ID of the associated message
+        externalFeedbackId:
+          type: integer
+          format: int32
+          description: ID of the external feedback (unique)
+        createdAt:
+          type: string
+          format: date-time
+        updatedAt:
+          type: string
+          format: date-time
       required:
-        - threadId
+        - id
         - messageId
-        - feedback
+        - externalFeedbackId
 
     ExternalFeedback:
       type: object
       properties:
         id:
-          type: string
+          type: integer
+          format: int32
+          description: Unique identifier for the feedback
         feedbackType:
           type: string
+          maxLength: 255
+          nullable: true
+          description: Type of the feedback
         feedbackText:
           type: string
-        threadId:
-          type: string
-        messageId:
-          type: string
+          nullable: true
+          description: Text content of the feedback
         userId:
```

Comment:
This is an outdated field and not actually used anymore. I just [updated the api to no longer return it](https://github.com/Melodi-fyi/llm-ops-vercel/pull/1105), so let's remvoe it from here please.